app/services/update_course_stats.rb

```diff
@@ -82,6 +84,11 @@ def log_update_progress(step)
                           extra: { logs: @sentry_logs }
   end
 
+  def should_update_categories?
+    previous_categories_update = @course.flags['previous_categories_update']
```

Comment:
I don't think this is the right approach. The decision of whether to update a category should be based on `Category#updated_at` (and probably doesn't need to happen inside `UpdateCourseStats`.
`Category.refresh_categories_for` should apply a query to `Course.categories` to find the ones that were updated more than a day ago, and just update those ones.

One additional consideration is to make sure that `updated_at` gets changed when this happens, even if the actual contents of the category didn't change. This might or might not require [`touch`](https://apidock.com/rails/ActiveRecord/Timestamp/touch).

Reply:
Done, used `touch` to update `updated_at` timestamp

Reply:
This method can be deleted now, right?

Reply:
Yes, deleted....forgot to do that